src/main/java/org/rocstreaming/roctoolkit/RocContext.java

```diff
@@ -60,7 +68,11 @@ public RocContext() throws Exception {
      * @throws Exception if there are not enough resources.
      */
     public RocContext(RocContextConfig config) throws IllegalArgumentException, Exception {
-        super(tryOpen(config), null, RocContext::close);
+        super(tryOpen(config), null, ptr -> {
+            LOGGER.log(Level.FINE, "roc_context closing, ptr=" + Long.toHexString(ptr));
+            close(ptr);
+            LOGGER.log(Level.FINE, "roc_context closed, ptr=" + Long.toHexString(ptr));
+        });
     }
```

Comment:
Consider extracting close lambda into a method (in context/sender/receiver). Otherwise it looks a bit strange that ctor does not have code for opening (it's extracted into tryOpen), but instead has code for closing.

Reply:
done. Maybe naming is not the best, offers are welcome :)